dynacred/src/byzcoin/transaction.ts

```diff
@@ -0,0 +1,63 @@
+import { Argument, ByzCoinRPC, ClientTransaction, Instruction } from "@dedis/cothority/byzcoin";
+import { AddTxResponse } from "@dedis/cothority/byzcoin/proto/requests";
+import ISigner from "@dedis/cothority/darc/signer";
+
+export class Transaction {
+    private instructions: Instruction[] = [];
+
+    constructor(protected bc: ByzCoinRPC) {
+    }
+
+    async send(signers: ISigner[][], wait = 0): Promise<[ClientTransaction, AddTxResponse]> {
+        const ctx = ClientTransaction.make(this.bc.getProtocolVersion(), ...this.instructions);
+        this.instructions = [];
```

Comment:
Can you explain more? I don't see what should be changed / answered / commented here.

Reply:
when sending a transaction and there is for example a network issue, one is unable to retry it as the instructions will be empty. that might be a strand behavior.

Reply:
OK, now I see your point. The `Transaction` is in fact a `TransactionBuilder`, so it's normal it resets the instructions after `send`. But `send` could have the following signature:

```
send(signers: ISigner[][], wait = 0): Promise<[ClientTransaction, Promise<AddTxResponse>]>
```

This would mean:
- good: the caller always gets the `ClientTransaction` and can decide if she wants to re-submit it
- bad: the caller needs to `await` one of the returned objects:
```
    const [ctx, atxPr] = await builder.send(signers);
    const atx = await atxPr;
```

If you can remove the `bad` part from above, I'd be very happy ;)